lib/rucio/core/rule_grouping.py

```diff
@@ -855,12 +863,27 @@ def __create_lock_and_replica(file, dataset, rule, rse_id, staging_area, availab
                                      name=file['name'],
                                      bytes_=file['bytes'],
                                      existing_replica=existing_replica,
-                                     state=LockState.OK)
+                                     state=LockState.OK if not staging_required else LockState.REPLICATING)
             if rse_id not in locks_to_create:
                 locks_to_create[rse_id] = []
             locks_to_create[rse_id].append(new_lock)
             locks[(file['scope'], file['name'])].append(new_lock)
-            return False
+            if not staging_required:
+                return False
+
+            transfers_to_create.append(create_transfer_dict(dest_rse_id=rse_id,
+                                                            request_type=RequestType.STAGEIN,
+                                                            scope=file['scope'],
+                                                            name=file['name'],
+                                                            rule=rule,
+                                                            lock=new_lock,
+                                                            bytes_=file['bytes'],
+                                                            md5=file['md5'],
+                                                            adler32=file['adler32'],
+                                                            ds_scope=dataset['scope'],
+                                                            ds_name=dataset['name'],
+                                                            lifetime=maximum_pin_lifetime if staging_required else None,
```

Comment:
This should probably be the the rule_lifetime, unless the rule_lifetime is bigger than maximum_pin_lifetime, in which case maximum_pin_lifetime is used.
The `else None` part is also not really possible in this case :-)